frontend/src/components/SignUp.jsx

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import { connect } from "react-redux";
+
+import proxy from "../proxy.js";
+import passwordCheck from "../helpers/passwordCheck.js";
+
+class UnconnectedSignUp extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: "",
+      password: "",
+      confirmPassword: "",
+      signUpError: ""
+    };
+  }
+  submitLogIn = event => {
+    event.preventDefault();
+    if (this.state.password !== this.state.confirmPassword) {
+      this.setState({
+        signUpError: "passwords did not match",
+        confirmPassword: ""
+      });
+      return;
+    }
+    if (!passwordCheck(this.state.password)) {
+      this.setState({
+        signUpError:
+          "Password must be 7 or more characters long and contain one lowercase letter, one number, on upper case letter"
+      });
+      return;
+    }
+    let data = new FormData();
+    data.append("username", this.state.username);
+    data.append("password", this.state.password);
+    console.log(this.state.username + this.state.password);
+    console.log(event);
+    fetch(proxy + "/signup", { method: "POST", body: data })
+      .then(res => {
+        return res.text();
+      })
+      .then(resBody => {
+        const parsedBody = JSON.parse(resBody);
+        if (!parsedBody.success) {
+          this.setState({ signUpError: parsedBody.result });
+          return;
+        } else {
+          console.log(parsedBody.result);
+          this.setState({ signUpError: "noError", username: "", password: "" });
+        }
+      });
```

Comment:
Speaking of abstraction: You could create an `api.js` on the client which wraps a bunch of these fetch calls. Something like:

```js
submitLogin = ev => {
  ev.preventDefault();

  postSignup(this.state.username, this.state.password)
    .then(() => this.setState({ signUpError: null, username: '', password: '' })
    .catch(err => this.setState({ signUpError: err });
}
```